lib/service/connect.go

```diff
@@ -160,6 +160,15 @@ func (process *TeleportProcess) connect(role types.SystemRole) (conn *Connector,
 		process.log.Infof("Connecting to the cluster %v with TLS client certificate.", identity.ClusterName)
 		clt, err := process.newClient(process.Config.AuthServers, identity)
 		if err != nil {
+			// In the event that a user is attempting to connect a machine to
+			// a different cluster it will give a cryptic warning about an
+			// unknown certificate authority. Unfortunately we cannot intercept
+			// this error as it comes from the http package before a request is
+			// made. So provide a more user friendly error as a hint of what
+			// they can do to resolve the issue.
+			if strings.Contains(err.Error(), "certificate signed by unknown authority") {
+				process.log.Error("Was this node already registered to a different cluster? To re-join the node, remove `/var/lib/teleport` and try again")
```

Comment:
Do we have any docs on this? I don't recall from the top of my head, but maybe we do? If so it would be worthwhile linking to those here, as "remove `/var/lib/teleport/`" is a rather destructive action; it would be good to provide some context or troubleshooting docs, if possible.

Reply:
Agreed, as far as I know there are no docs for this but I can make a note to add some. I was just following the notes in the issue and if a node is being switched to a different cluster it's probably safest to start over with the configuration and registration anyways.

Reply:
Maybe if we change the wording a bit this would be more clear.

Instead of "To re-join the node, remove..."

we could say "To join this node to a new cluster, remove..."